score-and-rank-contributors

```diff
@@ -0,0 +1,69 @@
+#! /usr/bin/env python
+
+# Import required libraries
+import argparse
+import math
+import json
+import re
+from git import Repo
+from collections import defaultdict
+
+# Define a function to parse command-line arguments
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--since', type=str, help='Start date in YYYY-MM-DD format')
+    parser.add_argument('--authors', type=str, help='Path to authors file')
+    parser.add_argument('projects', nargs='+', help='List of project directories')
+    return parser.parse_args()
+
+# Define a function to calculate scores
+def calculate_scores(projects, since_date, authors_map):
+    scores = defaultdict(int)
+    for project in projects:
+        repo = Repo(project)
+        for commit in repo.iter_commits(since=since_date):
+            email = commit.author.email
+            author = authors_map.get(email, email)
+            scores[author] += 1  # score for a commit
+            scores[author] += math.ceil(commit.stats.total['lines'] / 100)  # score for line changes
```

Comment:
Is `total` the sum of additions and deletions? Or the net additions (additions - deletions)? My intention is to use the former one.

Reply:
`total` is the sum of additions and deletions, referring to  [GitPython doc](https://gitpython.readthedocs.io/en/stable/reference.html?highlight=git%20stats#git.util.Stats.total)